src/common/http_client.rs

```diff
@@ -65,7 +65,27 @@ fn https_client(
 
     // Configure TLS root certificate and validation
     if let Some(tls_config) = tls_config {
-        builder = builder.add_root_certificate(https_client_ca_cert(tls_config.ca_cert.as_ref())?);
+        if let Some(ca_cert) = tls_config.ca_cert.clone() {
+            match https_client_ca_cert(ca_cert.as_ref()) {
+                Ok(ca_cert) => {
+                    builder = builder.add_root_certificate(ca_cert);
+                }
+                Err(err) => {
+                    // I think it might be Ok to not fail here, if root certificate is not found
+                    // There are 2 possible scenarios:
+                    //
+                    // 1. Server TLS is either not used or it uses some other CA certificate (like global one)
+                    // 2. Server TLS is using self-signed certificate, and we should have it.
+                    //
+                    // In first case, we don't need to load the CA certificate, everything will work in either case.
+                    // In second case, we should have the CA certificate, request will fail because of invalid certificate.
+                    //
+                    // So both scenarios work exactly the same way if we fail early or not.
+                    // Warning message is needed for easier debugging in case of second scenario.
```

Comment:
@generall I do prefer to fail here, just to make things a bit more explicit, especially since this is security critical.

Either a user configures the CA, or the user does not.

If the user does configure the CA, we fail if:
- we fail to read the file
- we fail to parse the CA

I'd consider both cases to be valid failures.

If it is by accident the user can adjust the configuration and remove the CA.

Wdyt?

Reply:
In ideal situation I would agree, but out existing configuration, unfrtunatelly, specified path by default. So if user wants to disable it, they should explicitly overwrite default value. It might be a problem for cloud and existing users

Reply:
What is the reason to fail, if we could verify the connection by other existing certificates?

Reply:
> So if user wants to disable it, they should explicitly overwrite default value. 

That makes a lot of sense. Let's keep it.